### PR TITLE
telemetry migration

### PR DESCRIPTION
**Pull request by edublancas**
closes https://github.com/ploomber/sklearn-evaluation/issues/105

I migrated the telemetry to use the new module in ploomber-core. this gives us more flexibility to log certain input parameters. furthermore, we can now share this logic in other projects.

Note: the PR looks a bit busy because I applied black formatting

cc @idomic

### CODE REVIEW

File: src/sklearn_evaluation/evaluator.py [hunk 1]
```diff
@@ -160,7 +158,7 @@ def html_serializable(self):
         """
         return EvaluatorHTMLSerializer(self)
 
-    @SKLearnEvaluationLogger.log(feature='report')
+    @tel_clf_eval.log_call()
```
Comment (idomic) [inline src/sklearn_evaluation/evaluator.py:161]: Why aren't we passing here feature/action?

File: src/sklearn_evaluation/plot/classification.py [hunk 2]
```diff
@@ -57,7 +60,7 @@ def __init__(self, a, b, target_names) -> None:
 
 
 class ConfusionMatrix(Plot):
-    @SKLearnEvaluationLogger.log(feature="plot", action="confusion-matrix-init")
+    @tel_cm.log_call(log_args=True, ignore_args={"y_true", "y_pred", "cm"})
```
Comment (idomic) [inline src/sklearn_evaluation/plot/classification.py:63]: I hope these are stored in the metadata object, else it'll make it hard to join both event types

File: src/sklearn_evaluation/plot/classification.py [hunk 3]
```diff
@@ -84,12 +87,12 @@ def __init__(self, y_true, y_pred, target_names=None, normalize=False, cm=None):
 
         _plot_cm(self.cm, cmap, ax, self.target_names, self.normalize)
 
-    @SKLearnEvaluationLogger.log(feature="plot", action="confusion-matrix-sub")
+    @tel_cm.log_call()
```
Comment (idomic) [inline src/sklearn_evaluation/plot/classification.py:90]: Same as above, missing action, we want to know what was used.

File: src/sklearn_evaluation/plot/classification.py [hunk 4]
```diff
@@ -141,7 +144,7 @@ def _confusion_matrix(y_true, y_pred, normalize):
     return cm
 
 
-@SKLearnEvaluationLogger.log(feature="plot")
+@tel_cm.log_call(action="function")
```
Comment (idomic) [inline src/sklearn_evaluation/plot/classification.py:147]: I see here the action is function, super generic, how would we know it was a confusion matrix?

File: src/sklearn_evaluation/plot/feature_ranking.py [hunk 5]
```diff
@@ -21,7 +21,7 @@
 from scipy.stats import shapiro
 from scipy.stats import spearmanr
 from scipy.stats import kendalltau as sp_kendalltau
-from sklearn_evaluation.telemetry import SKLearnEvaluationLogger
+from sklearn_evaluation.telemetry import telemetry
```
Comment (idomic) [inline src/sklearn_evaluation/plot/feature_ranking.py:24]: We probably want the module to be called _telemetry?

File: src/sklearn_evaluation/plot/metrics.py [hunk 6]
```diff
@@ -1,22 +1,19 @@
 import numpy as np
 from sklearn_evaluation.metrics import compute_at_thresholds
 from sklearn_evaluation.plot.util import set_default_ax
-from ..telemetry import SKLearnEvaluationLogger
+from sklearn_evaluation.telemetry import telemetry
```
Comment (idomic) [inline src/sklearn_evaluation/plot/metrics.py:4]: same, `sklearn_evaluation._telemetry`

File: src/sklearn_evaluation/tracker.py [hunk 7]
```diff
@@ -186,6 +187,9 @@ def __getitem__(self, key):
         return self._data[key]
 
 
+tel_tracker = telemetry.create_group("sqlitetracker")
```
Comment (idomic) [inline src/sklearn_evaluation/tracker.py:190]: Again, we're changing SQLiteTracker to sqlitetracker, we might see a big hit on the data unless we change it in both ends.

**Review by idomic — CHANGES_REQUESTED** [review]
Added a few comments, we need to keep the same convention or change the dashboard to accept these new events/data.
Something that would make this review way easier is keeping `'` vs changing to `"` and some other styling/whitespaces that was added (I assume it's automatic and not sure why git shows it). i.e instead of 22 changed files it'd be maybe 5 and I'm sure it's all on the dev IDE.

File: src/sklearn_evaluation/plot/classification.py (hunk 4 above)
Comment (idomic) [reply src/sklearn_evaluation/plot/classification.py:147]: I assume by the group? But still why call it function vs it's name

**Comment by idomic** [comment]
@edublancas any updates on this? let's try closing it.

**Comment by edublancas** [comment]
it's gonna be difficult to merge this since there's been so many changes, and there is little benefit to it. closing it

**Pull request closed without merging**